src/MSALWrapper/AuthFlow/BadAuthFlowImplementException.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+    using System;
+
+    /// <summary>
+    /// The bad authflow implementation exception.
+    /// </summary>
+    public class BadAuthFlowImplementException : Exception

```

Comment:
I think we need a different name.
1. `Bad` doesn't necessarily capture what's gone wrong here. _Why_ is it bad?
2. `Implement` (which should probably be `Implementation`) also doesn't seem right. I think it's not necessarily that the implementation is bad, but that the auth flow returned `null`.

@kyle-rader can remind us, but I think this just means that the flow failed, not that the implementation was bad.

Reply:
Renamed it to `NullTokenResultException`, which should be more descriptive.